src/python/content.cpp

```diff
@@ -1179,13 +1191,19 @@ content_methods(py::class_<T, std::shared_ptr<T>, ak::Content>& x) {
           .def("tojson",
                &tojson_string<T>,
                py::arg("pretty") = false,
-               py::arg("maxdecimals") = py::none())
+               py::arg("maxdecimals") = py::none(),
+               py::arg("nan_string") = nullptr,
+               py::arg("infinity_string") = nullptr,
+               py::arg("minus_infinity_string") = nullptr)
```

Comment:
I was looking for the place where this is called from Python. Since the strings are passed through everywhere as `char*`, something has to guarantee their lifetime through the duration of this call. I think this does it: I _think_ the arguments to this function in Python must be guaranteed to exist at least until the function returns a value, though I'm not 100% sure of this and can't think of a good way to ensure it.

Perhaps in the `tojson_string` and `tojson_file` functions, these strings should be copied to `std::strings` on the stack, which are kept in scope until the underlying functions finish, just for safety? Maybe I'm being too paranoid. We can come back to this.

Reply:
I think, we are safe here. According to [this doc](https://python-textbok.readthedocs.io/en/1.0/Variables_and_Scope.html), the `nan_string`, `infinity_string` and `minus_infinity_string` are accessible from the point at which they are defined in `tojson` or `fromjson` until the end of these functions, and exist for as long as the functions are executing.